run-tests-locally.sh

```diff
@@ -0,0 +1,37 @@
+#!/bin/bash
+set -eu
+
+if [ $# -ge 1 ] && [ -n "$1" ]
+then
+    echo "Running test container against $1"
+
+    export ECR_BASE_IMAGE="060113405249.dkr.ecr.eu-west-2.amazonaws.com/cic-cri-front-ecr-containerrepository-ymfhj103wnlw:de58a087b557fa09fd77c95024bb7481f91e3505:latest"
+    export SAM_STACK=$1 
+    export AWS_REGION="eu-west-2"
+    export TEST_REPORT_DIR="results"
+    export ENVIRONMENT="dev"
+    export DockerImageName="$SAM_STACK"_testcontainerimage
+
+    mkdir -p results # The directory on the host where the test results will be outputted.
+
+    aws cloudformation describe-stacks --stack-name "$SAM_STACK" --region "$AWS_REGION" --query 'Stacks[0].Outputs[].{key: OutputKey, value: OutputValue}' --output text > cf-output.txt
+    eval $(awk '{ printf("export CFN_%s=\"%s\"\n", $1, $2) }' cf-output.txt)
+    awk '{ printf("CFN_%s=\"%s\"\n", $1, $2) }' cf-output.txt > docker_vars.env
+    echo TEST_REPORT_DIR="$TEST_REPORT_DIR" >> docker_vars.env
+    echo TEST_REPORT_ABSOLUTE_DIR="$TEST_REPORT_DIR" >> docker_vars.env
+    echo TEST_ENVIRONMENT="$ENVIRONMENT" >> docker_vars.env
+    echo ENVIRONMENT="$ENVIRONMENT" >> docker_vars.env
+    echo SAM_STACK_NAME="$SAM_STACK" >> docker_vars.env
+    echo GITHUB_ACTIONS=true >> docker_vars.env
+
+    #clean existing container and image before creating a new one
+    echo "Removing existing containers and images for the test"
+    docker ps -a --filter "ancestor=$DockerImageName" --filter "status=exited" -q | xargs docker rm
+    docker images $DockerImageName -q |xargs docker rmi
```

Comment:
should we be doing this here if we are using `docker run --rm` ?

Reply:
This is to make sure we clean up before each run and make sure there are no images left behind form previous runs. For example failed run.